python/ray/data/_internal/dataset_logger.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import os
+from typing import Optional
+
+import ray
+from ray._private.ray_constants import LOGGER_FORMAT
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
+class DatasetLogger:
+    """Logger for Ray Datasets which, in addition to logging to stdout,
+    also writes to a separate log file at `DatasetLogger.DEFAULT_DATASET_LOG_PATH`.
+    """
+
+    DEFAULT_DATASET_LOG_PATH = "logs/datasets.log"
+
+    def __init__(self, log_name, dataset_log_path: Optional[str] = None):
+        """Initialize DatasetLogger for a given `log_name`.
+
+        Args:
+            log_name: Name of logger (usually passed into `logging.getLogger(...)`)
+            dataset_log_path: If specified, write logs to this path
+            instead of `DatasetLogger.DEFAULT_DATASET_LOG_PATH`
+        """
+        self.logger = logging.getLogger(log_name)
+        self.logger.setLevel(logging.INFO)
```

Comment:
this is not needed, right? why we need to set default log level here?

Reply:
i found that when creating a new logger here (potentially due to adding the new logging `Handler` / `Formatter`), it will not output logs to the external file if we do not first explicitly specify the logging level. i believe this is because in the `logging` library, if the level if not specified, it is set to `NOTSET`:
https://github.com/python/cpython/blob/main/Lib/logging/__init__.py#L1497

Reply:
@scottjlee - got it, could we add a comment in code? Thanks.